pkg/component/runtime/manager.go

```diff
@@ -705,6 +705,12 @@ func (m *Manager) update(model component.Model, teardown bool) error {
 		var stoppedWg sync.WaitGroup
 		stoppedWg.Add(len(stop))
 		for _, existing := range stop {
+			if existing.shuttingDown.Load() {
+				// it's already shutting down, move on
+				stoppedWg.Done()
+				continue
+			}
+			m.logger.Debugf("Stopping component %q", existing.id)
```

Comment:
Why do it here? Why not do it inside of `existing.stop(`? Seems safer, that way if the code is updated some where else to call `existing.stop(` the same logic would apply. Seems weird that it would be checked outside of the function that sets it.

Reply:
If you are referring to the log, it's there because that's where I landed when investigating the bug initially.
If you are referring to the stop() itself, the initial fix was only here in the manager without any modification to the service runtime object.
Would you prefer to have only for a service runtime object ?

Reply:
@pchila No I am referring to this line function https://github.com/elastic/elastic-agent/blob/main/pkg/component/runtime/runtime.go#L189 that is being called from https://github.com/elastic/elastic-agent/blob/main/pkg/component/runtime/manager.go#L708 (which is what your changing here).

This runtime wraps the actual managers. So this change will apply to all managers.

I thinking the function could easily be changed to:

```
func (s *componentRuntimeState) stop(teardown bool, signed *component.Signed) error {
	if s.shuttingDown.Load() {
                // already stopping
		return nil
        }
	s.shuttingDown.Store(true)
	if teardown {
		return s.runtime.Teardown(signed)
	}
	return s.runtime.Stop()
}
```

Reply:
done in 2606c30428f6e43099f7e262351d1cb2bd95570a